analysis/src/machine_check.rs

```diff
@@ -337,6 +337,18 @@ impl TypeChecker {
                                 Item::TypeDeclaration(enum_decl),
                             );
                         }
+                        asm::SymbolValue::TraitDeclaration(trait_decl) => {
+                            res.insert(
+                                ctx.clone().with_part(&name),
+                                Item::TraitDeclaration(trait_decl),
+                            );
+                        }
+                        asm::SymbolValue::TraitImplementation(trait_impl) => {
```

Comment:
I think we have to move this out of the `SymbolDefinition` item into its own ModuleStatement because it does not define a name.